src/lib.rs

```diff
@@ -121,29 +126,36 @@ pub fn send(channel: &UnixStream, wrapped: FdImplementor) -> Result<(), String>
 
     iov.iov_base = &mut fdtype as *mut i32 as *mut c_void;
     iov.iov_len = size_of_val(&fdtype);
-    
+
     message.msg_control = controlbuf.as_mut_ptr() as *mut c_void;
-    message.msg_controllen = size_of_val(&controlbuf);
+    message.msg_controllen = size_of_val(&controlbuf) as c_uint;
     message.msg_iov = &mut iov;
     message.msg_iovlen = 1;
 
     unsafe {
         let controlp : *mut libc::cmsghdr = message.msg_control as *mut libc::cmsghdr;
         (*controlp).cmsg_level = libc::SOL_SOCKET;
-        (*controlp).cmsg_type = SCM_RIGHTS;
-        (*controlp).cmsg_len = compute_msglen(size_of::<c_int>());
+        (*controlp).cmsg_type = get_SCM_RIGHTS();
+
+        #[cfg(not(target_os = "macos"))]
+        {
+            (*controlp).cmsg_len = compute_msglen(size_of::<c_int>()) as c_uint;
+        }
+        #[cfg(target_os = "macos")]
+        {
+            (*controlp).cmsg_len = size_of::<c_int>() as c_uint;
```

Comment:
I think this conditional check should be done for the compute_msglen function, rather than here. This would make the code simpler.